src/alire/alire-origins-deployers.adb

```diff
@@ -59,13 +59,23 @@ package body Alire.Origins.Deployers is
                                Folder : String) return Outcome
    is
       The_Deployer  : constant Deployer'Class := New_Deployer (From);
-      Deploy_Result : constant Outcome := The_Deployer.Deploy (Folder);
+      Result        : Outcome;
    begin
-      if Deploy_Result.Success then
-         return The_Deployer.Verify_Hashes (Folder);
-      else
-         return Deploy_Result;
+
+      --  1. Fetch sources
+      Result := The_Deployer.Fetch (Folder);
```

Comment:
Shouldn't we fetch and deploy in different directories?

Reply:
Do you see some advantage? Right now the tarball is deleted after unpacking so it doesn't mix with the actual sources.

Reply:
> Right now the tarball is deleted after unpacking so it doesn't mix with the actual sources.

That is the problem I had when trying to switch eagle_lander to archive distribution this weekend :)

So maybe the only argument left will be caching or the archives, not a huge deal at the moment...

Reply:
Yup, I hit that too when migrating the index. BTW, I have eagle_lander unmigrated for you to play with ;-) (in alire-project/alire-index#60)